fe/src/main/java/org/apache/doris/analysis/QueryStmt.java

```diff
@@ -285,27 +285,29 @@ protected void createSortInfo(Analyzer analyzer) throws AnalysisException {
         }
 
         sortInfo = new SortInfo(orderingExprs, isAscOrder, nullsFirstParams);
-        // order by w/o limit and offset in inline views, union operands and insert statements
+        // order by w/o limit and offset in inline views, set operands and insert statements
         // are ignored.
-        // TODO chenhao, open this when we don't limit rows subquery returns by SortNode.
-        /*if (!hasLimit() && !hasOffset() && !analyzer.isRootAnalyzer()) {
-         *   evaluateOrderBy = false;
-         *   // Return a warning that the order by was ignored.
-         *   StringBuilder strBuilder = new StringBuilder();
-         *   strBuilder.append("Ignoring ORDER BY clause without LIMIT or OFFSET: ");
-         *   strBuilder.append("ORDER BY ");
-         *   strBuilder.append(orderByElements.get(0).toSql());
-         *   for (int i = 1; i < orderByElements.size(); ++i) {
-         *       strBuilder.append(", ").append(orderByElements.get(i).toSql());
-         *   }
-         *   strBuilder.append(".\nAn ORDER BY appearing in a view, subquery, union operand, ");
-         *   strBuilder.append("or an insert/ctas statement has no effect on the query result ");
-         *   strBuilder.append("unless a LIMIT and/or OFFSET is used in conjunction ");
-         *   strBuilder.append("with the ORDER BY.");
-         * } else {
-         */
-        evaluateOrderBy = true;
-        //}
+        boolean allowIgnore = false;
+        if (ConnectContext.get() != null && ConnectContext.get().getSessionVariable().isAllowIgnoreOrderBy()) {
+            allowIgnore = true;
+        }
+        if (allowIgnore && !hasLimit() && !hasOffset() && !analyzer.isRootAnalyzer()) {
+            evaluateOrderBy = false;
+            // Return a warning that the order by was ignored.
+            StringBuilder strBuilder = new StringBuilder();
+            strBuilder.append("Ignoring ORDER BY clause without LIMIT or OFFSET: ");
+            strBuilder.append("ORDER BY ");
+            strBuilder.append(orderByElements.get(0).toSql());
+            for (int i = 1; i < orderByElements.size(); ++i) {
+                strBuilder.append(", ").append(orderByElements.get(i).toSql());
+            }
+            strBuilder.append(".\nAn ORDER BY appearing in a view, subquery, union operand, ");
+            strBuilder.append("or an insert/ctas statement has no effect on the query result ");
+            strBuilder.append("unless a LIMIT and/or OFFSET is used in conjunction ");
+            strBuilder.append("with the ORDER BY.");
```

Comment:
This `strBuilder` seems useless.

Reply:
`strBuilder` is mainly used in the `for` loop, and this code is consistent with impala, in case there is a need for merge mpala code in the future, it is also more convenient